src/group/mls_group/test_mls_group.rs

```diff
@@ -37,3 +43,267 @@ fn test_mls_group_persistence() {
+                MlsGroup::new_from_welcome_internal(broken_welcome, None, key_package_bundle)
+                    .expect_err("Creation of MLS group from a broken Welcome was successful.");
+
+            assert_eq!(
+                error,
+                WelcomeError::GroupSecretsDecryptionFailure(CryptoError::HpkeDecryptionError)
+            )
+        }
+    }
+}
+
+#[test]
+/// Test what happens if the KEM ciphertext for the receiver in the UpdatePath is broken.
+fn test_update_path() {
+    for ciphersuite in Config::supported_ciphersuites() {
+        // Basic group setup.
+        let group_aad = b"Alice's test group";
+
+        // Define credential bundles
+        let alice_credential_bundle =
+            CredentialBundle::new("Alice".into(), CredentialType::Basic, ciphersuite.name())
+                .unwrap();
+        let bob_credential_bundle =
+            CredentialBundle::new("Bob".into(), CredentialType::Basic, ciphersuite.name()).unwrap();
+
+        // Mandatory extensions
+        let capabilities_extension = Box::new(CapabilitiesExtension::new(
+            None,
+            Some(&[ciphersuite.name()]),
+            None,
+        ));
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        // Generate KeyPackages
+        let alice_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &alice_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+
+        let bob_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &bob_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+        let bob_key_package = bob_key_package_bundle.key_package();
+
+        // === Alice creates a group ===
+        let group_id = [1, 2, 3, 4];
+        let mut group_alice = MlsGroup::new(
+            &group_id,
+            ciphersuite.name(),
+            alice_key_package_bundle,
+            GroupConfig::default(),
+        )
+        .unwrap();
+
+        // === Alice adds Bob ===
+        let bob_add_proposal = group_alice.create_add_proposal(
+            group_aad,
+            &alice_credential_bundle,
+            bob_key_package.clone(),
+        );
+        let epoch_proposals = &[&bob_add_proposal];
+        let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) = group_alice
+            .create_commit(
+                group_aad,
+                &alice_credential_bundle,
+                epoch_proposals,
+                &[],
+                false,
+            )
+            .expect("Error creating commit");
+        let commit = match mls_plaintext_commit.content() {
+            MLSPlaintextContentType::Commit((commit, _)) => commit,
+            _ => panic!("Wrong content type"),
```

Comment:
Use `expect()` here as well.

Reply:
I don't think there is an existing function that either gives me a `Commit(..)` or an error. I'm using `match` here (and I copied this from another test btw.), so I can deconstruct the `MLSPlaintextContentType` and I'm not sure how I would use `expect()` here.

Reply:
True, we would need a function to extract a Commit from MLSPlaintext for that to work.